src/gamma.jl

```diff
@@ -337,40 +341,33 @@ function polygamma(m::Integer, z::Union{Float64,Complex{Float64}})
     # constants. We throw a DomainError() since the definition is unclear.
     real(m) < 0 && throw(DomainError())
 
-    s = m+1
+    s = Float64(m+1)
+    # It is safe to convert any integer (including `BigInt`) to a float here
+    # as underflow occurs before precision issues.
     if real(z) <= 0 # reflection formula
         (zeta(s, 1-z) + signflip(m, cotderiv(m,z))) * (-gamma(s))
     else
         signflip(m, zeta(s,z) * (-gamma(s)))
     end
 end
 
-# If we really cared about single precision, we could make a faster
-# Float32 version by truncating the Stirling series at a smaller cutoff.
-for (f,T) in ((:f32,Float32),(:f16,Float16))
-    @eval begin
-        zeta(s::Integer, z::Union{$T,Complex{$T}}) = $f(zeta(Int(s), f64(z)))
-        zeta(s::Union{Float64,Complex128}, z::Union{$T,Complex{$T}}) = zeta(s, f64(z))
-        zeta(s::Number, z::Union{$T,Complex{$T}}) = $f(zeta(f64(s), f64(z)))
-        polygamma(m::Integer, z::Union{$T,Complex{$T}}) = $f(polygamma(Int(m), f64(z)))
-        digamma(z::Union{$T,Complex{$T}}) = $f(digamma(f64(z)))
-        trigamma(z::Union{$T,Complex{$T}}) = $f(trigamma(f64(z)))
-    end
-end
+# TODO: better way to do this
```

Comment:
The `#TODO`  can be removed.
I believe it was in reference no the the `f64` etc.
But to the code that used that.
Which is what has been removed by my PR, and replaced with a "better way",
i,e, it is done.

Also I had (I thought) removed these definitions for `f64`, `f32` and `f16`,
for the definition on line 416.
(Which is restricted so one can only cast from the known, less precise types)
And with making use of `float`  more to find the partnered floating point type for a given integer type.
(Which works correctly for BigInts etc)